test/logoutresponse_test.rb

```diff
@@ -6,35 +6,38 @@
 class RubySamlTest < Minitest::Test
 
   describe "Logoutresponse" do
+
+    let(:valid_logout_response_without_settings) { OneLogin::RubySaml::Logoutresponse.new(valid_logout_response_document) }
+    let(:valid_logout_response) { OneLogin::RubySaml::Logoutresponse.new(valid_logout_response_document, settings) }
+    #let(:in_relation_to_request_id) { random_id }
+    #let(:valid_logout_response_with_responseid) { OneLogin::RubySaml::Logoutresponse.new(valid_logout_response_document{:uuid => in_relation_to_request_id}, settings) }
+
     describe "#new" do
       it "raise an exception when response is initialized with nil" do
         assert_raises(ArgumentError) { OneLogin::RubySaml::Logoutresponse.new(nil) }
       end
       it "default to empty settings" do
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new( valid_response)
-        assert_nil logoutresponse.settings
+        assert_nil valid_logout_response_without_settings.settings
       end
       it "accept constructor-injected settings" do
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, settings)
-        refute_nil logoutresponse.settings
+        refute_nil valid_logout_response.settings
       end
       it "accept constructor-injected options" do
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_response, nil, { :foo => :bar} )
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(valid_logout_response_document, nil, { :foo => :bar} )
         assert !logoutresponse.options.empty?
       end
       it "support base64 encoded responses" do
-        expected_response = valid_response
-        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(Base64.encode64(expected_response), settings)
-
-        assert_equal expected_response, logoutresponse.response
+        generated_logout_response = valid_logout_response_document
+        logoutresponse = OneLogin::RubySaml::Logoutresponse.new(Base64.encode64(generated_logout_response), settings)
+        assert_equal generated_logout_response, logoutresponse.response
       end
     end
 
-    describe "#validate" do
+    describe "#validate soft=true" do
```

Comment:
instead of having 2 `describe` blocks, you might want to use `soft = true` as `context`s inside `describe "#validate"`